rockcraft/services/lifecycle.py

```diff
@@ -45,14 +45,25 @@ def setup(self) -> None:
         # This inner import is necessary to resolve a cyclic import
         from rockcraft.services import RockcraftServiceFactory
 
+        # this will be removed once part_has_slices is added to craft-parts
+        def part_has_slices(data: dict[str, Any]) -> bool:
+            stage_packages = data.get("stage-packages", [])
+            return any(name for name in stage_packages if "_" in name)
+
         # Configure extra args to the LifecycleManager
         project = cast(Project, self._project)
         project_vars = {"version": project.version}
+        parts = project.parts
 
         services = cast(RockcraftServiceFactory, self._services)
         image_service = services.image
         image_info = image_service.obtain_image()
+        needs_chisel = any(part for part in parts if part_has_slices(parts[part]))
```

Comment:
I guess this will also slightly change after https://github.com/canonical/craft-parts/pull/637/

Reply:
yes, it will become `craft_parts.part_has_slices` instead.

Reply:
We can move detection and chisel installing to craft-parts, it already does that with other tools such as git.